interceptor/opencensus/opencensus.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ocinterceptor
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"google.golang.org/api/support/bundler"
+
+	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/census-instrumentation/opencensus-service/spanreceiver"
+)
+
+type OCInterceptor struct {
+	traceBundler     *bundler.Bundler
+	spanSink         spanreceiver.SpanReceiver
+	spanBufferPeriod time.Duration
+}
+
+func New(sr spanreceiver.SpanReceiver, opts ...OCOption) (*OCInterceptor, error) {
+	if sr == nil {
+		return nil, errors.New("needs a non-nil spanReceiver")
+	}
+	oci := &OCInterceptor{spanSink: sr}
+	for _, opt := range opts {
+		opt.WithOCInterceptor(oci)
+	}
+	return oci, nil
+}
+
+var _ agenttracepb.TraceServiceServer = (*OCInterceptor)(nil)
+
+var errUnimplemented = errors.New("unimplemented")
+
+func (oci *OCInterceptor) Config(tcs agenttracepb.TraceService_ConfigServer) error {
+	// TODO: Implement when we define the config receiver/sender.
+	return errUnimplemented
+}
+
+// Export is the gRPC method that receives streamed traces from
+// OpenCensus-traceproto compatible libraries/applications.
+func (oci *OCInterceptor) Export(tes agenttracepb.TraceService_ExportServer) error {
```

Comment:
I think there is a small problem here, Export can be called by multiple tasks in the same time, but in your current implementation you have only on bundler, here are the problems that you need to take care:
* Bundle has to be protected with a mutex if it is not thread-safe (gorutine safe) because it is shared between multiple calls.
* You need to add in the bundle {node, span[]} because if you receive from multiple nodes you need to keep the association between node and spans.

Alternatively (which probably is better) is to move bundle to be a local variable in the Export and avoid having it as a class member which cause all the problems I described.

Reply:
@bogdandrutu I removed the bundler field that was causing the confusion, my apologies. PTAL